libs/blocks/global-navigation/blocks/navDropdown/dropdown.js

```diff
@@ -0,0 +1,179 @@
+    const headingElem = itemTopParent.querySelector('h2');
+    itemTopParent.removeChild(headingElem);
+    itemTopParent.classList.add('feds-popup');
+
+    Array.prototype.forEach.call(itemTopParent.children, (section) => {
+      // TODO: allow link groups in small dropdowns too?
+      section.querySelectorAll('a').forEach((link, index) => {
+        link.setAttribute('daa-ll', getAnalyticsValue(link.textContent, index + 1));
+        link.classList.add('feds-navLink');
+      });
+    });
+
+    config.template.append(itemTopParent);
+  }
+
+  if (config.type === 'asyncDropdownTrigger') {
+    performance.mark('startAsyncDecoration');
+    const pathElement = config.item.querySelector('a');
+    if (!(pathElement instanceof HTMLElement)) return;
+    const path = localizeLink(pathElement.href);
+
+    const res = await fetch(`${path}.plain.html`);
+    if (res.status !== 200) return;
+    const content = await res.text();
+    const popupTemplate = toFragment`<div class="feds-popup">${content}</div>`;
+
+    // The resulting template structure should follow these rules:
+    // * a popup can have multiple columns;
+    // * a column can have multiple sections;
+    // * a section can have a headline and a collection of item(s)
+    Array.prototype.forEach.call(popupTemplate.children, (column) => {
+      const columnTemplate = toFragment`<div class="feds-popup-column"></div>`;
+      let columnSection = toFragment`<div class="feds-popup-section"></div>`;
+      let sectionItems;
+      let currIndex = 0;
+
+      while (column.children.length) {
+        const columnElem = column.firstElementChild;
+
+        if (columnElem.tagName === 'H5') {
+          // When encountering an h5, add the previous section to the column
+          if (columnSection.childElementCount) {
+            columnTemplate.append(columnSection);
+            currIndex = 0;
+          }
+
+          // Analysts requested no headings in the dropdowns,
+          // turning it into a simple div
+          const sectionHeadline = decorateHeadline(columnElem);
+          sectionItems = toFragment`<div class="feds-popup-items"></div>`;
+          columnSection = toFragment`<div class="feds-popup-section">
+              ${sectionHeadline}
+              ${sectionItems}
+            </div>`;
+        } else {
+          currIndex += 1;
+          let decoratedElem;
+
+          // Decorate link group
+          if (columnElem.classList.contains('link-group')) {
+            // TODO: check if links with just images also work
+            decoratedElem = decorateLinkGroup(columnElem, currIndex);
+          }
+
+          // Decorate CTA
+          if (columnElem.querySelector('strong > a')) {
+            decoratedElem = decorateCta({ elem: columnElem, index: currIndex });
+          }
+
+          // If element has been decorated, remove it
+          // from the initial list of column elements
+          if (decoratedElem) {
+            columnElem.remove();
```

Comment:
What happens with the `while` loop if the columnElement not a link-group, a CTA or UL?

Reply:
At the moment - nothing. But thinking of it, we should definitely allow simple links (not defined in a link group or a list). Otherwise, maybe we should just add the element unprocessed?

Reply:
Let's keep it as is for now. It's easier to add functionalityy in the future rather than taking it away and potentially breaking stuff.

Reply:
I retract my previous comment 😅 If the `columnElem` doesn't fit into a category where we know we need to add extra decoration, then we simply add it to the column/section as-is. However, it is very possible that a simple link is encountered, outside of a list, I will add the appropriate class and analytics attributes to it.